### ISSUE TITLE
DiscordWebhookClient spams rate limit warning

### ISSUE THREAD

**Issue by TehGM** [RogueException/Discord.Net#1472]
I am using DiscordWebhookClient to send a message to various channels. My Discord.Net package is version 2.1.1 (not preview).
For the most part it is fine, but problem starts with pre-emptive rate limit. The retry seems to be done in a loop, without really waiting for rate limits to expire. Using RetryMode.RetryRatelimit (for other cases I am using exceptions to handle manually).
I am listening to Log event, and then I use DataDog to aggregate structured logs, and my logs are spammed with this (each of the generated logs is for the exact same message):
![image](https://user-images.githubusercontent.com/9112561/78356830-402f0400-75a8-11ea-9087-20089c539158.png)
```
try
{
    using (DiscordWebhookClient client = new DiscordWebhookClient(target.Address))
    {
        if (onClientLog != null)
            client.Log += onClientLog;
        await client.SendMessageAsync(
            ping.Content,
            ping.IsTTS,
            ping.Embeds.Select(embed => embed.ConvertToEmbed()),
            ping.DisplayName,
            ping.DisplayAvatarUrl,
            new DiscordNet.RequestOptions()
            {
                CancelToken = cancellationToken,
                RetryMode = DiscordNet.RetryMode.RetryRatelimit
            });
        return true;
    }
}
catch (TimeoutException ex) when (LogWarningWithScope(logger, ex, "A request to Discord has timed out"))
{
    await Task.Delay(discordOptions.TimeoutRetryInterval);
    return false;
}
catch (Discord.Net.HttpException ex) when (LogWarningWithScope(logger, ex, "HttpException occured when making a request to Discord"))
{
    await Task.Delay(discordOptions.HttpExceptionRetryInterval);
    return false;
}
```

My Code tries to send the message only once (unless an exception has occured, which isn't the case here). The calling code for reference: 
```
uint attemptsCount = 0;
for (; ;)
{
    if (await DiscordWebhookSenderHelper.TrySendPingAsync(discordPing, target, _discordOptions.CurrentValue, _log, OnClientLog, cancellationToken))
        return;

    if (++attemptsCount >= _discordOptions.CurrentValue.MaxAttemptsCount)
    {
        _log.LogError("Sending message failed after {AttemptsCount} attempts. Sending aborted", attemptsCount);
        return;
    }
}
```

**Comment by BrammyS** [comment]
I'm having the same problem, but whit role creations.
Executing an API request form a different IP address will make the problem go away temporarily.
The only reason i can think of is that it's sending a lot of invalid requests resulting in a temporarily IP ban or something.

This is a really bad bug, since it's basically making my bot unusable at some times.

**Comment by quinchs** [comment]
Does this still happen in Discord.Net 3.0?

**Comment by MrCakeSlayer** [comment]
Closing due to lack of activity/responses. If you are still having this issue feel free to open a new issue. :)
